integrations/launchdarkly/client.py

```diff
@@ -0,0 +1,169 @@
+        kind = kind + "s"
+        url = kind if not resource_path else f"{kind}/{resource_path}"
+        url = url.replace("auditlogs", "auditlog")
+
+        params = {"limit": PAGE_SIZE}
+
+        while url:
+            try:
+                response = await self.send_api_request(
+                    endpoint=url, query_params=params
+                )
+                items = response.get("items", [])
+                yield items
+
+                if "_links" in response and "next" in response["_links"]:
+                    url = response["_links"]["next"]["href"]
+                else:
+                    total_count = response.get("totalCount")
+                    logger.info(f"Fetched {total_count} {kind} from Launchdarkly")
+                    break
+
+            except httpx.HTTPStatusError as e:
+                logger.error(
+                    f"HTTP error with status code: {e.response.status_code} and response text: {e.response.text}"
+                )
+                raise
+            except httpx.HTTPError as e:
+                logger.error(
+                    f"HTTP error occurred while fetching {kind} from LaunchDarkly: {e}"
+                )
+                raise
+
+    async def send_api_request(
+        self,
+        endpoint: str,
+        method: str = "GET",
+        query_params: Optional[dict[str, Any]] = None,
+        json_data: Optional[dict[str, Any]] = None,
+    ) -> dict[str, Any]:
+        logger.info(f"Requesting Launchdarkly data for endpoint: {endpoint}")
+        try:
+            endpoint = endpoint.replace("/api/v2/", "")
+            url = f"{self.api_url}/{endpoint}"
+            logger.info(
+                f"URL: {url}, Method: {method}, Params: {query_params}, Body: {json_data}"
+            )
+            response = await self.http_client.request(
+                method=method,
+                url=url,
+                params=query_params,
+                json=json_data,
+            )
+            response.raise_for_status()
+
+            logger.info(f"Successfully retrieved data for endpoint: {endpoint}")
+
+            return response.json()
+
+        except httpx.HTTPStatusError as e:
+            logger.error(
+                f"HTTP error on {endpoint}: {e.response.status_code} - {e.response.text}"
+            )
+            raise
+        except httpx.HTTPError as e:
+            logger.error(f"HTTP error on {endpoint}: {str(e)}")
+            raise
+
+    async def get_paginated_feature_flags(
+        self, kind: str
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        async for items in self.get_paginated_resource(kind=ObjectKind.PROJECT):
+            for project in items:
+                async for flags in self.get_paginated_resource(
+                    kind, resource_path=project["key"]
+                ):
+                    for flag in flags:
+                        flag.update({"__projectKey": project["key"]})
+                    yield flags

```

Comment:
This seem to be doing a lot thing due to the nested loops. Can you explain what you're trying to do we can simplify or boost the performance by running it in parallel?